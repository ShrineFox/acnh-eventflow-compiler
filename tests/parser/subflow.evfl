flow Root():
    run local_0p()
    run local_1p(Param=4)
    run local_2p(Param=4.3, X='123')
    run non::local_0p()
    run non::local_1p(Param=4)
    run non::local_2p(Param=4.3, X='123')
    return

flow local_0p():
    pass

flow local_1p(Param: int = 0):
    pass

flow local_2p(Param: float = 0.0, X: str = ''):
    pass
