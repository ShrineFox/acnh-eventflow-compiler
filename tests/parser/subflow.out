RootNode[name=Root, vardefs=[], out_edges=[Event0]]
RootNode[name=local_0p, vardefs=[], out_edges=[]]
RootNode[name=local_1p, vardefs=[RootNode.VarDef(name='Param', type=Type(type='int'), initial_value=0)], out_edges=[]]
RootNode[name=local_2p, vardefs=[RootNode.VarDef(name='Param', type=Type(type='float'), initial_value=0.0), RootNode.VarDef(name='X', type=Type(type='str'), initial_value='')], out_edges=[]]
SubflowNode[name=Event0, ns=, called_root_name=local_0p, params={}, out_edges=[Event1]]
SubflowNode[name=Event1, ns=, called_root_name=local_1p, params={'Param': TypedValue(type=Type(type='int'), value=4)}, out_edges=[Event2]]
SubflowNode[name=Event2, ns=, called_root_name=local_2p, params={'Param': TypedValue(type=Type(type='float'), value=4.3), 'X': TypedValue(type=Type(type='str'), value='123')}, out_edges=[Event3]]
SubflowNode[name=Event3, ns=non, called_root_name=local_0p, params={}, out_edges=[Event4]]
SubflowNode[name=Event4, ns=non, called_root_name=local_1p, params={'Param': TypedValue(type=Type(type='int'), value=4)}, out_edges=[Event5]]
SubflowNode[name=Event5, ns=non, called_root_name=local_2p, params={'Param': TypedValue(type=Type(type='float'), value=4.3), 'X': TypedValue(type=Type(type='str'), value='123')}, out_edges=[]]

